x/leverage/types/genesis.go

```diff
@@ -98,6 +103,15 @@ func (gs GenesisState) Validate() error {
 		}
 	}
 
+	if err := validateSpecialAssetPairDenoms(gs.SpecialPairs); err != nil {
+		return err
+	}
+	for _, pair := range gs.SpecialPairs {
+		if err := pair.Validate(); err != nil {
```

Comment:
After running `validateSpecialAssetPairDenoms` you execute `pair.Validate()` for every pair. You may want to move it all to the same function `validateSpecialAssetPairs` to avoid iterating twice.

Reply:
that works - combined them